lib/faraday_middleware/response_middleware.rb

```diff
@@ -47,7 +47,7 @@ def process_response(env)
     def parse(body)
       if self.class.parser
         begin
-          self.class.parser.call(body)
+          self.class.parser.call(body, @options)
```

Comment:
Rather than passing the entire `@options` hash, I would opt for deciding on a specific `@options` key (we can call it `parser_options`) to be passed to the parser.
This will make the call like:

```ruby
self.class.parser.call(body, @options[:parser_options])
```

IMPORTANT: We should pass this parameter ONLY if it's provided. We might have people defining custom parser that doesn't accept the second parameter. In order to maintain backward compatibility we need to make the second parameter optional